tokio-stream/src/stream_close.rs

```diff
@@ -0,0 +1,108 @@
+use crate::Stream;
+use pin_project_lite::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pin_project! {
+    /// Stream that returns None when the stream is exhausted. To be used with StreamMap to know when a stream ends.
+    /// # Example
+    /// Using `StreamNotifyClose` to handle closed streams with `StreamMap`
+    /// ```no_run
```

Comment:
Documentation examples should usually not be marked `no_run` if they contain a main function. In this case, the annotation hides the mistake where this example will panic if you run it.
```suggestion
    /// ```
```